pkg/network/node/pod.go

```diff
@@ -192,42 +192,34 @@ func (m *podManager) Start(rundir string, localSubnetCIDR string, clusterNetwork
 	return m.cniServer.Start(m.handleCNIRequest)
 }
 
-func (m *podManager) InitRunningPods(existingSandboxPods map[string]*kruntimeapi.PodSandbox, existingOFPodNetworks map[string]podNetworkInfo, cRunningPods []kapi.Pod) error {
+func (m *podManager) InitRunningPods(existingSandboxPods map[string]*kruntimeapi.PodSandbox, existingOFPodNetworks map[string]podNetworkInfo, cRunningPods map[string]kapi.Pod) error {
 	m.runningPodsLock.Lock()
 	defer m.runningPodsLock.Unlock()
-
-	for _, cPod := range cRunningPods {
-		cKey := getPodKey(cPod.Namespace, cPod.Name)
-
+	for _, sandbox := range existingSandboxPods {
+		cKey := getPodKey(sandbox.Metadata.Namespace, sandbox.Metadata.Name)
+		cPod, exists := cRunningPods[cKey]
+		if !exists {
+			glog.Warningf("No running pod for ready sandbox: %s with sandbox id: %s", cKey, sandbox.Id)
+			continue
+		}
 		var v1Pod v1.Pod
 		if err := kapiv1.Convert_core_Pod_To_v1_Pod(&cPod, &v1Pod, nil); err != nil {
 			glog.Warningf("Could not convert core pod: %s to v1Pod", cKey)
 			continue
 		}
-
 		podPortMapping := constructPodPortMapping(&v1Pod, net.ParseIP(v1Pod.Status.PodIP))
-
 		vnid, err := m.policy.GetVNID(cPod.Namespace)
```

Comment:
we still skip to portmap if we don;t find the GetNVID, is that correct?

Reply:
No, not anymore. We will open host ports and sync them in the step above in `shouldSyncHostports`, we won't add it do the running pods list though.

Reply:
Hmm, it's true that it might be better to have them aligned though and only open host ports and sync the pods that we add to `runningPods`

Reply:
I will correct that